helm/templates/secret-sidecar.yaml

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023 Cisco Systems, Inc. and/or its affiliates.
```

Comment:
Suggestion: Since this sets the same secret name as the one from secret.yaml, consider putting in in the same file (makes it clearer that it is the same object, but the content is conditional on the connection type). Also, some of the secret data is common (metadata, etc.) so we will avoid some duplication.